### PR TITLE
Fixes #16976 - add puppet env fact to spec helper

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: moduleroot/spec/spec_helper.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@
 add_custom_fact :mongodb_version, '2.4.14'          # puppetlabs-mongodb
 add_custom_fact :root_home, '/root'                 # puppetlabs-stdlib
 add_custom_fact :puppetversion, Puppet.version      # Facter, but excluded from rspec-puppet-facts
+add_custom_fact :puppet_environmentpath, Gem::Version.new(Puppet.version) >= Gem::Version.new('4.0') ? '/etc/puppetlabs/code/environments' : '/etc/puppet/environments' # added puppet env variable for puppet 3 and 4
```
Comment (stbenjam) [inline moduleroot/spec/spec_helper.rb]: The comment should so the source of the fact, which is `puppetlabs-stdlib`

**Review by stbenjam — CHANGES_REQUESTED** [review]
Otherwise OK

**Comment by chris1984** [comment]
@stbenjam ack fixed with adding puppetlabs-stdlib to the comment.

**Comment by stbenjam** [comment]
Just `puppetlabs-stdlib`, please, it should be consistent with the other facts, the comment is acting as a column that only has the source of the fact

**Comment by chris1984** [comment]
@stbenjam no problem, i have adjusted it like the others # puppetlabs-stdlib

**Review by stbenjam — APPROVED** [review]

**Comment by stbenjam** [comment]
Thanks! Looks ok to me, I'll test and sync the rest of the modules with this tomorrow hopefully

**Comment by chris1984** [comment]
@stbenjam cool thank you, once that gets pulled in I think it will help my other PR that is failing on the puppet 4 tests.

**Comment by stbenjam** [comment]
Thanks, confirmed it fixes puppet-capsule: https://travis-ci.org/stbenjam/puppet-capsule/builds/168628514

**Comment by stbenjam** [comment]
ACK

**Comment by ekohl** [comment]
@stbenjam I see you created a lot of modulesync PRs. I can ACK them all individually but when the tests pass you can just merge them.

**Comment by stbenjam** [comment]
@ehelms Went through and marked them all approved, I know it's always a bit of a hassle. Doesn't foreman just usually push to master?

**Comment by ekohl** [comment]
It's not that often that we sync so there's no formal procedure. What I think I did the last time:
- Push all branches to your own forks
- Create PRs for each branch (`hub pull-request` is what I use)
- Create a PR with a checklist linking to all the PRs
- Merge the individual PRs (checking them off in the modulesync PR)
- Merge the modulesync PR

I think this is at least partially scriptable but I haven't because it's not very often that this is done.

**Pull request merged**